java/arcs/android/storage/service/FutureRegistrationCallback.kt

```diff
@@ -0,0 +1,16 @@
+package arcs.android.storage.service
+
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.Future
+
+class FutureRegistrationCallback private constructor(
```

Comment:
KDoc needed

Reply:
Also, I'm curious why you chose to use `Future` here, instead of `Deferred`?

Reply:
It's used in non-suspending methods.

Reply:
As you saw below :-)